app/models/manageiq/providers/vmware/infra_manager/vm/operations/guest.rb

```diff
@@ -3,37 +3,45 @@ module ManageIQ::Providers::Vmware::InfraManager::Vm::Operations::Guest
 
   included do
     supports :reboot_guest do
-      unsupported_reason_add(:reboot_guest, unsupported_reason(:control)) unless supports?(:control)
+      reason = unsupported_reason(:control)
+      return reason if reason
+
       if current_state == "on"
         if tools_status == 'toolsNotInstalled'
-          unsupported_reason_add(:reboot_guest, _("The VM tools is not installed"))
+          _("The VM tools is not installed")
         end
       else
-        unsupported_reason_add(:reboot_guest, _("The VM is not powered on"))
+        _("The VM is not powered on")
       end
     end
 
     supports :shutdown_guest do
-      unsupported_reason_add(:shutdown_guest, unsupported_reason(:control)) unless supports?(:control)
+      reason = unsupported_reason(:control)
+      return reason if reason
```

Comment:
@kbrock I'm running some tests with rails 7... probably unrelated... I'm seeing a local jump error here because we're returning from within a block.

```
[----] E, [2024-07-01T16:07:26.262770#61590:7288] ERROR -- evm: [LocalJumpError]: unexpected return  Method:[block (2 levels) in <class:LogProxy>]
[----] E, [2024-07-01T16:07:26.262895#61590:7288] ERROR -- evm: /Users/joerafaniello/.gem/ruby/3.1.5/bundler/gems/manageiq-providers-vmware-5c67447bd206/app/models/manageiq/providers/vmware/infra_manager/vm/operations/guest.rb:20:in `block (2 levels) in <module:Guest>'
/Users/joerafaniello/Code/manageiq/app/models/mixins/supports_feature_mixin.rb:113:in `instance_eval'
/Users/joerafaniello/Code/manageiq/app/models/mixins/supports_feature_mixin.rb:113:in `check_supports'
/Users/joerafaniello/Code/manageiq/app/models/mixins/supports_feature_mixin.rb:71:in `supports?'
/Users/joerafaniello/Code/manageiq-ui-classic/app/helpers/application_helper/button/generic_feature_button_with_disable.rb:5:in `disabled?'
/Users/joerafaniello/Code/manageiq-ui-classic/app/helpers/application_helper/button/basic.rb:54:in `calculate_properties'
```

I guess we want a structure like

```ruby
if reason
...
elsif current_state == "on"
else
...
end
```

Can you take a look?